external_services/data.py

```diff
@@ -0,0 +1,74 @@
+import json
+import os
+import urllib.parse
+
+import requests
+
+import config
+from config import Config
+
+
+def api_call(endpoint: str, method: str = "GET", params: dict = None):
+    if params:
+        params = {k: v for k, v in params.items() if v is not None}
+    if config.__dict__.get("USE_LOCAL_DATA"):
+        return local_api_call(endpoint, params, method)
+    else:
+        if method:
+            if method == "POST":
+                return requests.post(endpoint, json=params)
+            elif method == "GET":
+                req = requests.PreparedRequest()
+                req.prepare_url(endpoint, params)
+                return requests.get(req.url)
+
+
+def get_data(endpoint: str, params: dict = None):
+    if params:
+        params = {k: v for k, v in params.items() if v is not None}
+    if config.__dict__.get("USE_LOCAL_DATA"):
+        return local_api_call(endpoint, params, "get")
+    else:
+        req = requests.PreparedRequest()
+        req.prepare_url(endpoint, params)
+        response = requests.get(req.url)
+        if response.status_code == 200:
+            return response.json()
+
+
+def post_data(endpoint: str, params: dict = None):
+    if params:
+        params = {k: v for k, v in params.items() if v is not None}
+    if config.__dict__.get("USE_LOCAL_DATA"):
+        return local_api_call(endpoint, params, "post")
+    else:
+        response = requests.post(endpoint, json=params)
+        if response.status_code in [200, 201]:
+            return response.json()
+
+
+def local_api_call(endpoint: str, params: dict = None, method: str = "get"):
+    api_data_json = os.path.join(
+        Config.FLASK_ROOT,
+        "tests",
+        "api_data",
+        method.lower() + "_endpoint_data.json",
```

Comment:
i wonder if this should be in an envvar?

Reply:
Which bit? This is only used in development/local environments isn't it?

Reply:
i suppose, the bigger question is probably should we store all our file paths in config so we go to one place to change them rather than app code

Reply:
Got it. I've pulled this out into config and added a test to confirm this works